src/bin/pgcopydb/ld_stream.c

```diff
@@ -35,6 +35,150 @@
 static bool updateStreamCounters(StreamContext *context,
 								 LogicalMessageMetadata *metadata);
 
+/*
+ * Determines whether a message needs to be filtered out based on the
+ * provided filters. The caller function should have already parsed and
+ * validated the filters.
+ */
+bool
+ShouldFilterOutMessage(SourceFilters *filters, char *nspname, char *relname)
+{
+	/*
+	 * Validate nspname is not NULL or empty
+	 */
+	if (nspname == NULL || strlen(nspname) == 0)
+	{
+		log_error("BUG: nspname is NULL or empty");
+		return false;
+	}
+
+	/*
+	 * Validate relname is not NULL or empty
+	 */
+	if (relname == NULL || strlen(relname) == 0)
+	{
+		log_error("BUG: relname is NULL or empty");
+		return false;
+	}
+
+	/*
+	 * Filter-out pgcopydb.* in any case
+	 */
+	if (strcmp(nspname, "pgcopydb") == 0)
```

Comment:
Use streq instead of strcmp across the change.

```suggestion
	if (streq(nspname, "pgcopydb"))
```